hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -155,7 +155,8 @@ protected void handlePutDirEvent(OmDirectoryInfo directoryInfo,
   protected void handleDeleteKeyEvent(OmKeyInfo keyInfo,
                                       Map<Long, NSSummary> nsSummaryMap)
       throws IOException {
-    long parentObjectId = keyInfo.getParentObjectID();
+    NSSummary currentNSSummary = nsSummaryMap.get(keyInfo.getObjectID());
```

Comment:
Not very clear why we have taken parent Id from NSSummary object based. What was the issue in earlier code if we directly take parentObjectId from OmKeyInfo here ?